src/tool/hpcrun/audit/auditor.c

```diff
@@ -237,8 +243,13 @@ static void hook_open(uintptr_t* cookie, struct link_map* map) {
       }
     } else abort();
   }
-  entry->start = (void*)map->l_addr + start;
-  entry->end = (void*)map->l_addr + end;
+
+  // load module.  we must adjust it by start so that it has the proper
+  // base address value for the subsequent calculations.
+  if (ao_flags & AO_VDSO) map->l_addr -= (long) start;
+
+  entry->start = (void*)map->l_addr + (long) start;
+  entry->end = (void*)map->l_addr + (long) end;
```

Comment:
Last comment: is there a reason these are `long` and not `intptr_t` (or better `uintptr_t`)? `long` is not necessarily large enough to hold a pointer-sized value.

Otherwise everything looks fine.

Reply:
change to intptr_t. the arithmetic didn't work out with uintptr_t.